azkaban-web-server/src/main/resources/azkaban/webapp/servlet/velocity/scheduledflowpage.vm

```diff
@@ -114,17 +114,27 @@
                 #else
                   <td>Not Applicable</td>
                 #end
-                <td><button type="button" data-toggle="modal" data-target="#executionOptions">show</button></td>
+                <td>
+                ## Changed the style of "Show" button to be consistent with other buttons.
+                  <button type="button" class="btn btn-sm btn-info" data-toggle="modal"
+                          data-target="#executionOptions-${velocityCount}">Show
+                  </button>
+                </td>
 
-                  <div class="modal fade" id="executionOptions" tabindex="-1" role="dialog" aria-labelledby="executionOptionsLabel">
+              ## Distinguished executionOptions of each schedule with unique count number to fix
+              ## the issue of same display for all schedules
+                <div class="modal fade" id="executionOptions-${velocityCount}" tabindex="-1"
+                     role="dialog"
+                     aria-labelledby="executionOptionsLabel">
                       <div class="modal-dialog" role="document">
                           <div class="modal-content">
                               <div class="modal-header">
                                   <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                                   <h4 class="modal-title" id="executionOptionsLabel">Execution Options</h4>
                               </div>
                               <div class="modal-body">
-                                  ${sched.executionOptions.toObject()}
+                              ## Used <pre> to display text with code format
+                                <pre>${sched.executionOptions.toJSON()}</pre>
```

Comment:
I suggest keeping the template code as simple as possible i.e. no function calls within ${}.

How about moving this logic into java code?

Reply:
The only way I can think of is to first convert the executionOptions to JSON for each schedule in the servlet call azkaban.webapp.servlet.ScheduleServlet#handleGetAllSchedules. Then return a new list of Objects which contain the converted results instead of the original schedules. I agree it's ideal to keep the template simple, but the additional complexity to the logic might not worth it.